src/main/java/uk/gov/ons/ctp/integration/rhsvc/cloud/FirestoreDataStore.java

```diff
@@ -0,0 +1,187 @@
+package uk.gov.ons.ctp.integration.rhsvc.cloud;
+
+import com.godaddy.logging.Logger;
+import com.godaddy.logging.LoggerFactory;
+import com.google.api.core.ApiFuture;
+import com.google.cloud.firestore.DocumentReference;
+import com.google.cloud.firestore.FieldPath;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.QueryDocumentSnapshot;
+import com.google.cloud.firestore.QuerySnapshot;
+import com.google.cloud.firestore.WriteResult;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.ons.ctp.common.error.CTPException;
+import uk.gov.ons.ctp.common.error.CTPException.Fault;
+
+@Service
+public class FirestoreDataStore implements CloudDataStore {
+  private static final Logger log = LoggerFactory.getLogger(FirestoreDataStore.class);
+
+  // Names of environment variables which firestore uses for connection information
+  public static final String FIRESTORE_PROJECT_ENV_NAME = "GOOGLE_CLOUD_PROJECT";
+
+  @Autowired private Firestore firestore;
+
+  /**
+   * Write object to Firestore collection. If the object already exists then it will be overwritten.
+   *
+   * @param schema - holds the name of the collection that the object will be added to.
+   * @param key - identifies the object within the collection.
+   * @param value - is the object to be written to Firestore. It must provide get methods that allow
+   *     access to its contents.
+   * @throws CTPException if any failure was detected interacting with Firestore.
+   */
+  @Override
+  public void storeObject(final String schema, final String key, final Object value)
+      throws CTPException {
+    log.debug("Saving object to Firestore. Schema '{}' with key '{}'", schema, key);
+
+    // Store the object
+    ApiFuture<WriteResult> result = firestore.collection(schema).document(key).set(value);
+
+    // Wait for Firestore to complete
+    try {
+      result.get();
+      log.debug("Firstore save completed. Schema '{}' with key '{}'", schema, key);
```

Comment:
typo

Reply:
A Firstore is a seasonal shop, often a garage forecourt, selling a specific type of tree

Reply:
Fixed.